go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -540,6 +544,14 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 			config.Metrics.NodeServerCert, config.Metrics.NodeServerPrivKey, stopChan, wg)
 	}
 
+	if runMode.ovnkubeController {
+		err := <-ovnkubeControllerStartChan
+		if err != nil {
+			return err
+		}
+		defer cm.Stop()
```

Comment:
very good point!
I tried to make what you suggest, but that looks very ugly, because we abuse conditional defers during the startup, so it is much less changes (and chances to break something) to just move a single `Start` call to another goroutine, (and you can find that defer dance in this version, but at least it fixes the problem)

Reply:
LGTM, I am fine with what you have here. Usually though you just need a bit more of syntactic sugar to make things less ugly, making things symmetric usually helps as well.